app/models/user.rb

```diff
@@ -3,4 +3,21 @@ class User < ActiveRecord::Base
   # , :lockable, :timeoutable and :omniauthable
   devise :database_authenticatable, :registerable, :confirmable,
          :recoverable, :rememberable, :trackable, :validatable
+
+  has_many :pending_purchases
+
+  def add_to_cart(book)
+    pending_purchases.create!(book_id: book.id,
+                              price_at_purchase: book.price,
+                              quantity: 1
+                             )
+  end
+
+  def remove_from_cart(book)
+    pending_purchases.find_by_book_id(book.id).destroy
+  end
+
+  def in_cart?(book)
+    pending_purchases.any? { |h| h[:book_id] == book.id }
```

Comment:
I think it would probably be better as `pending_purchase`. Just because that's very clear. Brevity isn't necessary with code.